blockindex/bloomfilterindexer.go

```diff
@@ -0,0 +1,174 @@
+// Copyright (c) 2020 IoTeX Foundation
+// This is an alpha (internal) release and is not suitable for production. This source code is provided 'as is' and no
+// warranties are given as to title or non-infringement, merchantability or fitness for purpose and, to the extent
+// permitted by law, all liability for your use of the code is disclaimed. This source code is governed by Apache
+// License 2.0 that can be found in the LICENSE file.
+
+package blockindex
+
+import (
+	"context"
+	"sync"
+
+	"github.com/iotexproject/go-pkgs/bloom"
+
+	"github.com/iotexproject/iotex-core/action"
+	"github.com/iotexproject/iotex-core/blockchain/block"
+	"github.com/iotexproject/iotex-core/blockchain/blockdao"
+	"github.com/iotexproject/iotex-core/db"
+	"github.com/iotexproject/iotex-core/pkg/util/byteutil"
+	"github.com/pkg/errors"
+)
+
+const (
+	// BloomFilterNamespace indicates the kvstore namespace to store RangeBloomFilters
+	BloomFilterNamespace = "RangeBloomFilters"
+	// CurrentHeightKey indicates the key of current bf indexer height in underlying DB
+	CurrentHeightKey = "CurrentHeight"
+)
+
+type (
+	// BloomFilterIndexer is the interface for bloomfilter indexer
+	BloomFilterIndexer interface {
+		blockdao.BlockIndexer
+		RangeBloomFilterSize() uint64
+		RangeBloomFilter(uint64) (bloom.BloomFilter, error)
```

Comment:
Comments should be added to public struct/interface, so we need to move the comments up to here.